backend/src/journey/entity/legDetails.entity.ts

```diff
@@ -1,61 +1,71 @@
-import { ApiPropertyOptional } from "@nestjs/swagger";
-import { IsArray, IsEnum, IsInstance, IsNumber, IsOptional } from "class-validator";
-import { LegRealtimeTripStatus } from "../../vrr/entity/legRealtimeTripStatus.entity";
+import { ApiProperty, ApiPropertyOptional } from "@nestjs/swagger";
+import { IsArray, IsInstance, IsInt, IsOptional, Min } from "class-validator";
+import { LocationCoordinates } from "../../locationFinder/entity/locationCoordinates.entity";
+import { JourneyLocation } from "./journeyLocation.entity";
 import { LegInfo } from "./legInfo.entity";
+import { LegInterchange } from "./legInterchange.entity";
 
+/**
+ * Details of journey leg.
+ */
 export class LegDetails {
 
-  @IsNumber()
-  @IsOptional()
-  @ApiPropertyOptional({
-    description: "Distance of leg in meter?",
-    type: Number
-  })
-  distance?: number;
-
-  @IsNumber()
-  @IsOptional()
-  @ApiPropertyOptional({
+  @IsInt()
+  @Min(0)
+  @ApiProperty({
     description: "Duration of leg in seconds.",
     type: Number,
+    required: true,
     example: 480
   })
-  duration?: number;
+  duration!: number;
 
   @IsArray()
   @IsInstance(LegInfo, { each: true })
-  @IsOptional()
-  @ApiPropertyOptional({
-    description: "Leg information.",
-    type: [LegInfo]
+  @ApiProperty({
+    description: "Information like restrictions regarding this specifically dated leg (can be empty).",
+    type: [LegInfo],
+    required: true
   })
-  infos?: LegInfo[];
+  infos!: LegInfo[];
```

Comment:
I get the idea behind the unionisation. I'm not sure whether we should do this or we shouldn't. Normally this separation enables changes to the internal types without having to modify the external ones. But maybe we could still split them up for that later.

Reply:
I don´t fully understand how the exclamation mark and the constructors are correlated (and redundant)? (However no problem to fix that one or the other way.)

Reply:
What exactly do you mean with "unionisation" of dtos and entities? Is it about creating new types by extending others?

Reply:
I think we don't need the constructor right now, because in legDetailsMapper.service.ts you are using the duck typing capabilities of typescript: As long as an object has the same interface as a class, it can be used wherever the class is expected. However I don't think that will enforce the checks added using the decorators - which might also be another reason to separate entities and DTOs...

Reply:
> What exactly do you mean with "unionisation" of dtos and entities? Is it about creating new types by extending others?

Yes. But you didn't only create a new type. You basically made them the same. But as I said, I'm still unsure how strong the separation should be myself.

Reply:
As far as I am aware the only place where DTOs and entities are being mixed is `LocationCoordinatesDto` (extends `LocationCoordinates`). I suggest to implement that class manually (even if it means some redundancy for now) to clearly seperate DTOs and entities. Furthermore in all the entities I remove the exclamation marks and constructors.

Reply:
@timvahlbrock Is this resolved from your point of you with the changes I made?

Reply:
Yes, sorry, oversaw the notification.